hoomd/comm.py

```diff
@@ -379,3 +380,29 @@ def barrier(self):
         if _hoomd.is_MPI_available():
             self.cpp_mpi_conf.barrier()
 
+    @contextlib.contextmanager
+    def localize_abort(self):
+        """ Localize MPI_Abort to this partition.
+
+        HOOMD calls MPI_Abort to tear down all running MPI processes whenever
+        there is an uncaught exception. By default, this will abort the entire
+        MPI execution. When using partitions (``nrank is not None``), an uncaught
+        exception on one partition will therefore abort all of them.
+
+        Use the return value of :py:meth:`localize_abort()` as a context manager
+        to tell HOOMD that all operations within the context will use only
+        that MPI communicator so that an uncaught exception in one partition
+        will only abort that partition and leave the others running.
+        """
+
+        global _current_communicator
+        prev = _current_communicator
+
+        _current_communicator = self
+        yield None
+        _current_communicator = prev
+
+# store the "current" communicator to be used for MPI_Abort calls. This defaults
+# to the world communicator, but users can opt in to a more specific
+# communicator using the GPU.localize_abort context manager
```

Comment:
```suggestion
# communicator using the Device.localize_abort context manager
```

Reply:
Why Device? I thought your suggestion was to add the GPU specific method to only the GPU class.

Reply:
If the user has multiple GPU's through an MPI simulation, they can't selectively abort? I know that even when running on the GPU you require a CPU, but the user wouldn't have access to a CPU device as far as I am aware.

Reply:
Nevermind, I was confusing `localize_abort` and `enable_profiling`